### PR TITLE
add base user service

### PR DESCRIPTION
**Pull request by dlimx**
## Notes

Add the db connection and an API endpoint to get user

Adds ability to use `async`/`await` for db actions

## Testing

Host locally and Get the `/user/:id` endpoint

### CODE REVIEW

File: models/db.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const mysql = require('mysql');
+const util = require('util');
+
+const pool = mysql.createPool({
+    host: process.env.DB_HOST || 'fakehost',
```
Comment (DomPSU) [inline models/db.js:5]: Should we remove the || 'fakehost' and the other three || 'fake'? I looked at the mysql documentation and it looks like its not standard? I am unsure

File: models/schema.sql [hunk 2]
```diff
@@ -13,10 +13,10 @@ DROP TABLE IF EXISTS Users;
 CREATE TABLE Users(
 	userID INT AUTO_INCREMENT NOT NULL PRIMARY KEY,
     userLevelID INT NOT NULL,
-    name VARCHAR(50),
+    name VARCHAR(50) NOT NULL,
```
Comment (DomPSU) [inline models/schema.sql:16]: Should any other attributes be NOT NULL? Or were you just working on Users?

File: models/db.js (hunk 1 above)
Comment (dlimx) [reply models/db.js:5]: We probably want to use `.env` to supply these at all times anyways, no preference either way, I can remove

File: models/schema.sql (hunk 2 above)
Comment (dlimx) [reply models/schema.sql:16]: I think basically all of the attributes on all our tables should be `NOT NULL`, but let's discuss within the team

File: models/db.js (hunk 1 above)
Comment (DomPSU) [reply models/db.js:5]: Yea lets remove it. Sorry for originally saying to add it.

File: models/schema.sql (hunk 2 above)
Comment (DomPSU) [reply models/schema.sql:16]: I saw the comment in discord. I vote for making all NOT NULL.

Comment (mckerracher) [reply models/schema.sql:16]: Yeah, I agree. I think all of those attributes are/should be necessary.

**Comment by dlimx** [comment]
Updated PR with `NOT NULL` constraint for all fields (except `Product.descript`)

**Pull request merged**